src/client/java/dev/enjarai/trickster/screen/SpellPartWidget.java

```diff
@@ -274,34 +276,56 @@ protected void stopDrawing() {
             } else {
                 setSubPartInTree(drawingPart, Optional.of(newPart), spellPart, false);
             }
+        } else if (compiled.equals(EXPAND_TO_OUTER_CIRCLE_GLYPH)) {
+            if (drawingPart != spellPart) {
+                boolean b = spellPart.glyph == drawingPart;
+                setSubPartInTree(drawingPart, Optional.of(drawingPart), spellPart, true);
+
+                if (b) {
+                    drawingPart = spellPart;
+                }
+            }
```

Comment:
If `drawingPart == spellPart`, you can skip the recursive search entirely to save some processing.